File: practica1/P1S3/Pescado.cpp
#include "Comida.cpp"
#include <iostream>
using namespace std;
class Pescado : public Comida
{    
    public:
        Pescado() :Comida()
        {
            precio = 3.0;
        }

        double getPrecio()
        {
            return precio;
        }

        void setPrecio(double p)
        {
            precio = p;
        }

        string toString()
        {
            string resultado = "El pescado cuesta " + to_string(precio);
            return resultado;
        }
};

File: practica1/P1S3/Java/src/practica1dss3/Pescado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package practica1dss3;

/**
 *
 * @author USUARIO
 */
public class Pescado implements Comida{
    private double precio;
    private boolean hayStock;
    
    public Pescado(){
        precio = 3.0;
        hayStock = true;
    }
    
    @Override
    public double getPrecio() {
        return precio;
    }

    @Override
    public void setPrecio(double p) {
        precio = p;
    }
    
    @Override
    public String toString(){
        if(hayStock)
            return("El pescado cuesta " + getPrecio());
        else
            return("Ya no queda pescado");
    }

    @Override
    public void cambioStock() {
        hayStock = false;
    }
}
